plugin-server/src/worker/ingestion/event-pipeline.ts

```diff
@@ -0,0 +1,84 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { runInstrumentedFunction } from '../../main/utils'
+import { Hub } from '../../types'
+import { runOnAction, runOnEvent, runOnSnapshot, runProcessEvent } from '../plugins/run'
+import { Action } from './../../types'
+import { ingestEvent } from './ingest-event'
+
+export async function runEventPipeline(server: Hub, event: PluginEvent): Promise<void> {
+    const processedEvent = await processEvent(server, event)
+
+    if (processedEvent) {
+        let actionMatches: Action[] = []
+        await Promise.all([
+            runInstrumentedFunction({
+                server,
+                event: processedEvent,
+                func: async (event) => {
+                    const result = await ingestEvent(server, event)
+                    actionMatches = result.actionMatches || []
```

Comment:
maybe for a follow up refactor but will just leave the comment here so I don't forget - would be great to separate out the action handling code into its own function rather than `ingestEvent` if we're doing this pipeline approach

Reply:
See comment above.